androidshared/src/main/java/org/odk/collect/androidshared/ui/OnBackPressedKeyListener.kt

```diff
@@ -0,0 +1,17 @@
+package org.odk.collect.androidshared.ui
+
+import android.content.DialogInterface
+import android.view.KeyEvent
+
+class OnBackPressedKeyListener(private val onBackPressed: () -> Unit) : DialogInterface.OnKeyListener {
+    override fun onKey(dialog: DialogInterface?, keyCode: Int, event: KeyEvent?): Boolean {
+        event?.let {
+            if (it.action == KeyEvent.ACTION_DOWN && it.keyCode == KeyEvent.KEYCODE_BACK) {
```

Comment:
This (surprisingly) feels like the cleanest way to achieve what we want here. The only other option I could think of is to use `setCanceledOnTouchOutside` and `setOnCancelListener` but we that's much more awkward to test ("canceledOnTouchOutside" isn't exposed as state anywhere).